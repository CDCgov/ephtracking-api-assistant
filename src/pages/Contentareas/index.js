import React, { Component, Fragment } from 'react';

import CodeBlock from '../../components/CodeBlock';
import Preview from '../../components/Preview';

class Contentareas extends Component {
  constructor(props) {
    super(props);    
    this.handleInputChange = this.handleInputChange.bind(this);
  }


  handleInputChange(event) {
    const { name, checked } = event.target;
    this.setState({
      [name]: checked
    });
  }

  render() {
    const url = `https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/json`
    return (
      <Fragment>
        <h1 className="title">Retrieving List of all Content Areas</h1>
        <h5 className="title is-5">Usage</h5>
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/{'{'}version{'}'}/contentareas/{'{'}returnType{'}'}[?apiToken]
        </CodeBlock>
        <hr />
        <h5 className="title is-5">Set parameters</h5>        
        <hr />
        <Preview url={url} />
      </Fragment>
    );
  }
}

export default Contentareas;
