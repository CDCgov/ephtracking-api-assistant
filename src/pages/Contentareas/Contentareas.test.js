import React from 'react';
import { shallow } from 'enzyme';

import Contentareas from './';

describe('Page Contentareas', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Contentareas />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handle getChildMeasure check', () => {
    const wrapper = shallow(<Contentareas />);
    const event = {
      target: { type: 'checkbox', checked: true, name: 'getChildMeasure' }
    };
    wrapper.instance().handleInputChange(event);
    expect(wrapper.state()).toMatchObject({ getChildMeasure: true });
  });

  it('handle getMultiMeasure check', () => {
    const wrapper = shallow(<Contentareas />);
    const event = {
      target: { type: 'checkbox', checked: true, name: 'getMultiMeasure' }
    };
    wrapper.instance().handleInputChange(event);
    expect(wrapper.state()).toMatchObject({ getMultiMeasure: true });
  });
});
