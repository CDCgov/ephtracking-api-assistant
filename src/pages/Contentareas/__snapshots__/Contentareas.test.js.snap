// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Contentareas renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contentareas />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Retrieving List of all Content Areas
</h1>,
        <h5
          className="title is-5"
>
          Usage
</h5>,
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/
          {
          version
          }
          /contentareas/
          {
          getChildMeasure
          }
          /
          {
          getMultiMeasure
          }
          /
          {
          returnType
          }
          [?apiToken]
</CodeBlock>,
        <hr />,
        <h5
          className="title is-5"
>
          Set parameters
</h5>,
        <div
          className="field"
>
          <label
                    className="checkbox"
          >
                    <input
                              checked={false}
                              className="input-checkbox"
                              name="getChildMeasure"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    getChildMeasure
          </label>
</div>,
        <div
          className="field"
>
          <label
                    className="checkbox"
          >
                    <input
                              checked={false}
                              className="input-checkbox"
                              name="getMultiMeasure"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    getMultiMeasure
          </label>
</div>,
        <hr />,
        <Preview
          url="https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/0/0/json"
          validUrl={true}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retrieving List of all Content Areas",
          "className": "title",
        },
        "ref": null,
        "rendered": "Retrieving List of all Content Areas",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Usage",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Usage",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/contentareas/",
            "{",
            "getChildMeasure",
            "}",
            "/",
            "{",
            "getMultiMeasure",
            "}",
            "/",
            "{",
            "returnType",
            "}",
            "[?apiToken]",
          ],
        },
        "ref": null,
        "rendered": Array [
          "https://ephtracking.cdc.gov/apigateway/api/",
          "{",
          "version",
          "}",
          "/contentareas/",
          "{",
          "getChildMeasure",
          "}",
          "/",
          "{",
          "getMultiMeasure",
          "}",
          "/",
          "{",
          "returnType",
          "}",
          "[?apiToken]",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Set parameters",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Set parameters",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className="checkbox"
>
            <input
                        checked={false}
                        className="input-checkbox"
                        name="getChildMeasure"
                        onChange={[Function]}
                        type="checkbox"
            />
            getChildMeasure
</label>,
          "className": "field",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className="input-checkbox"
                name="getChildMeasure"
                onChange={[Function]}
                type="checkbox"
/>,
              "getChildMeasure",
            ],
            "className": "checkbox",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": "input-checkbox",
                "name": "getChildMeasure",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "getChildMeasure",
          ],
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className="checkbox"
>
            <input
                        checked={false}
                        className="input-checkbox"
                        name="getMultiMeasure"
                        onChange={[Function]}
                        type="checkbox"
            />
            getMultiMeasure
</label>,
          "className": "field",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className="input-checkbox"
                name="getMultiMeasure"
                onChange={[Function]}
                type="checkbox"
/>,
              "getMultiMeasure",
            ],
            "className": "checkbox",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": "input-checkbox",
                "name": "getMultiMeasure",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "getMultiMeasure",
          ],
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/0/0/json",
          "validUrl": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Retrieving List of all Content Areas
</h1>,
          <h5
            className="title is-5"
>
            Usage
</h5>,
          <CodeBlock>
            https://ephtracking.cdc.gov/apigateway/api/
            {
            version
            }
            /contentareas/
            {
            getChildMeasure
            }
            /
            {
            getMultiMeasure
            }
            /
            {
            returnType
            }
            [?apiToken]
</CodeBlock>,
          <hr />,
          <h5
            className="title is-5"
>
            Set parameters
</h5>,
          <div
            className="field"
>
            <label
                        className="checkbox"
            >
                        <input
                                    checked={false}
                                    className="input-checkbox"
                                    name="getChildMeasure"
                                    onChange={[Function]}
                                    type="checkbox"
                        />
                        getChildMeasure
            </label>
</div>,
          <div
            className="field"
>
            <label
                        className="checkbox"
            >
                        <input
                                    checked={false}
                                    className="input-checkbox"
                                    name="getMultiMeasure"
                                    onChange={[Function]}
                                    type="checkbox"
                        />
                        getMultiMeasure
            </label>
</div>,
          <hr />,
          <Preview
            url="https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/0/0/json"
            validUrl={true}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retrieving List of all Content Areas",
            "className": "title",
          },
          "ref": null,
          "rendered": "Retrieving List of all Content Areas",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Usage",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Usage",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "https://ephtracking.cdc.gov/apigateway/api/",
              "{",
              "version",
              "}",
              "/contentareas/",
              "{",
              "getChildMeasure",
              "}",
              "/",
              "{",
              "getMultiMeasure",
              "}",
              "/",
              "{",
              "returnType",
              "}",
              "[?apiToken]",
            ],
          },
          "ref": null,
          "rendered": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/contentareas/",
            "{",
            "getChildMeasure",
            "}",
            "/",
            "{",
            "getMultiMeasure",
            "}",
            "/",
            "{",
            "returnType",
            "}",
            "[?apiToken]",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Set parameters",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Set parameters",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="checkbox"
>
              <input
                            checked={false}
                            className="input-checkbox"
                            name="getChildMeasure"
                            onChange={[Function]}
                            type="checkbox"
              />
              getChildMeasure
</label>,
            "className": "field",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="input-checkbox"
                  name="getChildMeasure"
                  onChange={[Function]}
                  type="checkbox"
/>,
                "getChildMeasure",
              ],
              "className": "checkbox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "input-checkbox",
                  "name": "getChildMeasure",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "getChildMeasure",
            ],
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="checkbox"
>
              <input
                            checked={false}
                            className="input-checkbox"
                            name="getMultiMeasure"
                            onChange={[Function]}
                            type="checkbox"
              />
              getMultiMeasure
</label>,
            "className": "field",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="input-checkbox"
                  name="getMultiMeasure"
                  onChange={[Function]}
                  type="checkbox"
/>,
                "getMultiMeasure",
              ],
              "className": "checkbox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "input-checkbox",
                  "name": "getMultiMeasure",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "getMultiMeasure",
            ],
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/0/0/json",
            "validUrl": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
